kyo-prelude/shared/src/test/scala/kyo/BatchTest.scala

```diff
@@ -767,4 +767,87 @@ class BatchTest extends Test:
         }
     }
 
+    "source effects handled before Batch.run" - {
```

Comment:
Checks that computations after the source call have the correct pending effects that can be handled before `Batch.run`.